### PR TITLE
Merge nixos-hardware into nixpkgs under `/nixos/hardware`

### PR DESCRIPTION
**Pull request by Profpatsch**
Merges https://github.com/NixOS/nixos-hardware into nixpkgs.

This will give better discoverability, and considering the low
turnover (less than 100 commits in the last year and only 350 total)
it won’t make any dent on the size of nixpkgs.

We have a monorepo, let’s use it.

cc @Mic92 and @yegortimoshenko as the top contributors to nixos-hardware.

Followup:
- [ ] Move the few issues on nixos-hardware to the nixpkgs issue tracker
- [ ] Archive nixos-hardware
- [ ] Move the stray hardware definitions in `nixos/modules/installer/cd-dvd` (e.g. the raspberripy definitions) to `nixos/hardware`
- [ ] Think about how to integrate the tests
- [ ] Integrate into nixos-generate-config
- [ ] Mention in nixos installer configuration.

### CODE REVIEW

File: nixos/hardware/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+language: nix
```
Comment (yegortimoshenko) [inline nixos/hardware/.travis.yml:1]: Would make sense to have this file removed, I think.

File: nixos/hardware/COPYING [hunk 2]
```diff
@@ -0,0 +1,111 @@
+Statement of Purpose
```
Comment (yegortimoshenko) [inline nixos/hardware/COPYING:1]: This is a public domain license. So, for simplicity's sake, I'd remove it and keep a top-level MIT license.

**Review by lukateras — APPROVED** [review]
I think this is a great idea.

File: nixos/hardware/.travis.yml (hunk 1 above)
Comment (Profpatsch) [reply nixos/hardware/.travis.yml:1]: Yeah, we can think about how to integrate it into the existing tests later.

File: nixos/hardware/COPYING (hunk 2 above)
Comment (Profpatsch) [reply nixos/hardware/COPYING:1]: I wasn’t sure of that, as IANAL I opted for keeping it safe. I would make that decision in a separate PR.

**Comment by Mic92** [comment]
I am also all for it. It also makes it easier to apply changes that depend on specific nixpkgs revisions.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -55,6 +55,15 @@
 # NixOS integration test driver
 /nixos/lib/test-driver  @tfc
 
+# NixOS hardware descriptions
+/nixos/hardware/beagleboard/pocketbeagle @yegortimoshenko
+/nixos/hardware/dell/xps/13-9370 @moredread
```
Comment (Mic92) [inline .github/CODEOWNERS:60]: cc @Moredread

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -55,6 +55,15 @@
...
+/nixos/hardware/dell/xps/13-9380 @kalbasit
```
Comment (Mic92) [inline .github/CODEOWNERS:61]: @kalbasit

**Review by Mic92 — APPROVED** [review]

**Comment by grahamc** [comment]
I don't think this is a good idea. These profiles aren't tested regularly, and merging them in to nixpkgs means there should be a quality standard IMO. They're also fairly opinionated in ways that our users may not want. I don't this is a healthy move for Nixpkgs or for the nixos-hardware project.`

**Comment by grahamc** [comment]
I'm also not sure about the license. Nixpkgs' expressions are all MIT. I think this should be reverted for more consideration.

**Comment by edolstra** [comment]
Indeed, changes to the scope of Nixpkgs require an RFC.

**Comment by cleverca22** [comment]
seems this can also sometimes confuse `git bisect`:
```
[clever@amd-nixos:~/apps/nixpkgs]$ git bisect bad
Bisecting: 284 revisions left to test after this (roughly 8 steps)
[c3b2ac63ff06e5dd4fa66f75bfab7892b3edfdc5] Add deepsleep to avoid wakeups during sleep mode (#136)

[clever@amd-nixos:~/apps/nixpkgs]$ nix-build -A platformio && ls -l result/
attribute 'platformio' in selection path 'platformio' not found

[clever@amd-nixos:~/apps/nixpkgs]$ nix repl .
Welcome to Nix version 3.0pre20200829_f156513. Type :? for help.

Loading '.'...
Added 0 variables.

nix-repl> :q

[clever@amd-nixos:~/apps/nixpkgs]$ cat default.nix
# This file is necessary so nix-env -qa does not break,
# when nixos-hardware is used as a channel
{}

[clever@amd-nixos:~/apps/nixpkgs]$
```

bisect randomly switched over to commits from nixos-hardware, and then `nix-build` fails due to an empty set

**Pull request merged**